dashboard/config/routes.rb

```diff
@@ -233,7 +233,7 @@
 
   post 'level_assets/upload', to: 'level_assets#upload'
 
-  resources :level_starter_assets, only: [:show], param: 'level_name' do
+  resources :level_starter_assets, only: [:show], param: 'level_name', constraints: {level_name: /[^\/]+/} do
```

Comment:
This change will cause some breaking changes to `level_starter_assets#file` because it uses `params[:format]`, but should be easily fixable:

https://github.com/code-dot-org/code-dot-org/blob/faa85aaa126254e264293c7483715caeb2d901b6/dashboard/app/controllers/level_starter_assets_controller.rb#L21-L31

Reply:
we will want line 24 above to change to something like `friendly_name = "#{params[:filename]}}" `, which ends up reading much better than the original!

Reply:
Thank you! Good catch - is the format needed for anything as far as you know or is it okay to drop it from the friendly name?

Reply:
I was rushing to get this in an out-of-band deploy tonight, but StarLabs decided to postpone, so FWIW, I'll also add some tests here.